SU2_CFD/include/numerics/NEMO/NEMO_diffusion.hpp

```diff
@@ -0,0 +1,257 @@
+   */
+  void GetViscousProjFlux(su2double *val_primvar,
+                          su2double **val_gradprimvar,
+                          su2double *val_eve,
+                          const su2double *val_normal,
+                          su2double *val_diffusioncoeff,
+                          su2double val_viscosity,
+                          su2double val_therm_conductivity,
+                          su2double val_therm_conductivity_ve,
+                          CConfig *config);
+  /*!
+   * \brief TSL-Approximation of Viscous NS Jacobians for arbitrary equations of state.
+   * \param[in] val_Mean_PrimVar - Mean value of the primitive variables.
+   * \param[in] val_gradprimvar - Mean value of the gradient of the primitive variables.
+   * \param[in] val_Mean_SecVar - Mean value of the secondary variables.
+   * \param[in] val_laminar_viscosity - Value of the laminar viscosity.
+   * \param[in] val_eddy_viscosity - Value of the eddy viscosity.
+   * \param[in] val_thermal_conductivity - Value of the thermal conductivity.
+   * \param[in] val_heat_capacity_cp - Value of the specific heat at constant pressure.
+   * \param[in] val_dist_ij - Distance between the points.
+   * \param[in] val_normal - Normal vector, the norm of the vector is the area of the face.
+   * \param[in] val_dS - Area of the face between two nodes.
+   * \param[in] val_Proj_Visc_Flux - Pointer to the projected viscous flux.
+   * \param[out] val_Proj_Jac_Tensor_i - Pointer to the projected viscous Jacobian at point i.
+   * \param[out] val_Proj_Jac_Tensor_j - Pointer to the projected viscous Jacobian at point j.
+   */
+  void GetViscousProjJacs(su2double *val_Mean_PrimVar,
+                          su2double **val_Mean_GradPrimVar,
+                          su2double *val_Mean_Eve,
+                          su2double *val_Mean_Cvve,
+                          su2double *val_diffusion_coeff,
+                          su2double val_laminar_viscosity,
+                          su2double val_thermal_conductivity,
+                          su2double val_thermal_conductivity_ve,
+                          su2double val_dist_ij, su2double *val_normal,
+                          su2double val_dS, su2double *val_Fv,
+                          su2double **val_Jac_i, su2double **val_Jac_j,
+                          CConfig *config);
+  /*!
+   * \brief Compute the viscous flow residual using an average of gradients.
+   * \param[out] val_residual - Pointer to the total residual.
+   * \param[out] val_Jacobian_i - Jacobian of the numerical method at node i (implicit computation).
+   * \param[out] val_Jacobian_j - Jacobian of the numerical method at node j (implicit computation).
+   * \param[in] config - Definition of the particular problem.
+   */
+  void ComputeResidual(su2double *val_residual,
+                       su2double **val_Jacobian_i,
+                       su2double **val_Jacobian_j,
+                       CConfig *config);
```

Comment:
You should add `override` / `final` specifiers to the new numerics classes and their methods. See for example `roe.hpp` for how they are used.